### DISCUSSION TITLE
Git detached head

### DISCUSSION THREAD

**Discussion by SAC020** [vladmandic/sdnext#3403] [General]
Am on the dev branch and I am seeing some "git detached head" messages in the log. I don't know if it's an issue, so just asking this in a discussion thread - is it something normal / expected?

thanks

> 
> PS C:\ai\automatic> .\webui.bat --medvram --debug
> Using VENV: C:\ai\automatic\venv
> 08:22:55-179048 INFO     Starting SD.Next
> 08:22:55-182040 INFO     Logger: file="C:\ai\automatic\sdnext.log" level=DEBUG size=65 mode=create
> 08:22:55-184036 INFO     Python version=3.11.9 platform=Windows bin="C:\ai\automatic\venv\Scripts\python.exe"
>                          venv="C:\ai\automatic\venv"
> 08:22:55-378508 INFO     Version: app=sd.next updated=2024-09-02 hash=bba17766 branch=dev
>                          url=https://github.com/vladmandic/automatic/tree/dev ui=dev
> 08:22:56-191172 INFO     Latest published version: bab17a0b4f91b41c885f10262ef8c8e70ba72faa 2024-08-31T20:57:34Z
> 08:22:56-204169 INFO     Platform: arch=AMD64 cpu=Intel64 Family 6 Model 165 Stepping 5, GenuineIntel system=Windows
>                          release=Windows-10-10.0.22631-SP0 python=3.11.9
> 08:22:56-206160 DEBUG    Setting environment tuning
> 08:22:56-207130 INFO     HF cache folder: C:\Users\sebas\.cache\huggingface\hub
> 08:22:56-209125 DEBUG    Torch allocator: "garbage_collection_threshold:0.80,max_split_size_mb:512"
> 08:22:56-219929 DEBUG    Torch overrides: cuda=False rocm=False ipex=False diml=False openvino=False
> 08:22:56-221134 DEBUG    Torch allowed: cuda=True rocm=True ipex=True diml=True openvino=True
> 08:22:56-232221 INFO     nVidia CUDA toolkit detected: nvidia-smi present
> 08:22:56-313452 WARNING  Modified files: ['models/Reference/playgroundai--playground-v2-1024px-aesthetic.jpg']
> 08:22:56-403214 INFO     Verifying requirements
> 08:22:56-407228 INFO     Verifying packages
> 08:22:56-450259 DEBUG    Repository update time: Tue Sep  3 01:08:32 2024
> 08:22:56-450821 INFO     Startup: standard
> 08:22:56-452117 INFO     Verifying submodules
**> 08:22:59-888555 DEBUG    Git detached head detected: folder="extensions-builtin/sd-extension-chainner" reattach=main
> 08:22:59-889846 DEBUG    Submodule: extensions-builtin/sd-extension-chainner / main
> 08:23:00-009633 DEBUG    Git detached head detected: folder="extensions-builtin/sd-extension-system-info" reattach=main
> 08:23:00-011627 DEBUG    Submodule: extensions-builtin/sd-extension-system-info / main
> 08:23:00-132141 DEBUG    Git detached head detected: folder="extensions-builtin/sd-webui-agent-scheduler" reattach=main
> 08:23:00-133139 DEBUG    Submodule: extensions-builtin/sd-webui-agent-scheduler / main
> 08:23:00-301094 DEBUG    Git detached head detected: folder="extensions-builtin/sdnext-modernui" reattach=dev
> 08:23:00-302121 DEBUG    Submodule: extensions-builtin/sdnext-modernui / dev
> 08:23:00-444978 DEBUG    Git detached head detected: folder="extensions-builtin/stable-diffusion-webui-rembg"
>                          reattach=master
> 08:23:00-445976 DEBUG    Submodule: extensions-builtin/stable-diffusion-webui-rembg / master
> 08:23:00-567390 DEBUG    Git detached head detected: folder="modules/k-diffusion" reattach=master
> 08:23:00-568388 DEBUG    Submodule: modules/k-diffusion / master
> 08:23:00-686117 DEBUG    Git detached head detected: folder="wiki" reattach=master
> 08:23:00-687142 DEBUG    Submodule: wiki / master**
> 08:23:00-759576 DEBUG    Register paths
> 08:23:00-846371 DEBUG    Installed packages: 207
> 08:23:00-847608 DEBUG    Extensions all: ['Lora', 'sd-extension-chainner', 'sd-extension-system-info',
>                          'sd-webui-agent-scheduler', 'sd-webui-controlnet', 'sdnext-modernui',
>                          'stable-diffusion-webui-images-browser', 'stable-diffusion-webui-rembg']
> 08:23:01-025059 DEBUG    Running extension installer:
>                          C:\ai\automatic\extensions-builtin\sd-extension-system-info\install.py
> 08:23:01-393054 DEBUG    Running extension installer:
>                          C:\ai\automatic\extensions-builtin\sd-webui-agent-scheduler\install.py
> 08:23:01-757647 DEBUG    Running extension installer: C:\ai\automatic\extensions-builtin\sd-webui-controlnet\install.py
> 08:23:02-266898 DEBUG    Running extension installer:
>                          C:\ai\automatic\extensions-builtin\stable-diffusion-webui-images-browser\install.py
> 08:23:02-632486 DEBUG    Running extension installer:
>                          C:\ai\automatic\extensions-builtin\stable-diffusion-webui-rembg\install.py
> 08:23:02-996279 DEBUG    Extensions all: []
> 08:23:02-997230 INFO     Extensions enabled: ['Lora', 'sd-extension-chainner', 'sd-extension-system-info',
>                          'sd-webui-agent-scheduler', 'sd-webui-controlnet', 'sdnext-modernui',
>                          'stable-diffusion-webui-images-browser', 'stable-diffusion-webui-rembg']
> 08:23:02-999228 INFO     Verifying requirements
> 08:23:03-000225 DEBUG    Setup complete without errors: 1725340983
> 08:23:03-006918 DEBUG    Extension preload: {'extensions-builtin': 0.0, 'extensions': 0.0}
> 08:23:03-007886 DEBUG    Starting module: <module 'webui' from 'C:\\ai\\automatic\\webui.py'>
> 08:23:03-008883 INFO     Command line args: ['--medvram', '--debug'] medvram=True debug=True
> 08:23:03-009881 DEBUG    Env flags: []
> 08:23:08-808284 INFO     Load packages: {'torch': '2.4.0+cu124', 'diffusers': '0.31.0.dev0', 'gradio': '3.43.2'}
> 08:23:09-836090 DEBUG    Read: file="config.json" json=35 bytes=1455 time=0.000
> 08:23:09-838085 DEBUG    Unknown settings: ['cross_attention_options']
> 08:23:09-840080 INFO     Engine: backend=Backend.DIFFUSERS compute=cuda device=cuda attention="Scaled-Dot-Product"
>                          mode=no_grad
> 08:23:09-898922 INFO     Device: device=NVIDIA GeForce RTX 4080 n=1 arch=sm_90 cap=(8, 9) cuda=12.4 cudnn=90100
>                          driver=560.81
> 08:23:09-900917 DEBUG    Read: file="html\reference.json" json=52 bytes=29118 time=0.001
> 08:23:10-300475 DEBUG    ONNX: version=1.19.0 provider=CUDAExecutionProvider, available=['AzureExecutionProvider',
>                          'CPUExecutionProvider']
> 08:23:10-478064 DEBUG    Importing LDM
> 08:23:10-493584 DEBUG    Entering start sequence
> 08:23:10-496353 DEBUG    Initializing
> 08:23:10-521259 INFO     Available VAEs: path="models\VAE" items=0
> 08:23:10-523255 DEBUG    Available UNets: path="models\UNET" items=0
> 08:23:10-524901 DEBUG    Available T5s: path="models\T5" items=0
> 08:23:10-525871 INFO     Disabled extensions: ['sd-webui-controlnet', 'sdnext-modernui']
> 08:23:10-527865 DEBUG    Read: file="cache.json" json=2 bytes=10089 time=0.000
> 08:23:10-534874 DEBUG    Read: file="metadata.json" json=470 bytes=1620174 time=0.005
> 08:23:10-539834 DEBUG    Scanning diffusers cache: folder=models\Diffusers items=0 time=0.00
> 08:23:10-541198 INFO     Available models: path="models\Stable-diffusion" items=20 time=0.01
> 08:23:10-730721 DEBUG    Load extensions
> 08:23:10-778022 INFO     Extension: script='extensions-builtin\Lora\scripts\lora_script.py'
>                          [2;36m08:23:10-775030[0m[2;36m [0m[34mINFO    [0m LoRA networks: [33mavailable[0m=[1;36m70[0m
>                          [33mfolders[0m=[1;36m2[0m
> 08:23:11-151765 INFO     Extension: script='extensions-builtin\sd-webui-agent-scheduler\scripts\task_scheduler.py' Using
>                          sqlite file: extensions-builtin\sd-webui-agent-scheduler\task_scheduler.sqlite3
> 08:23:11-346834 DEBUG    Extensions init time: 0.61 sd-webui-agent-scheduler=0.33
>                          stable-diffusion-webui-images-browser=0.18
> 08:23:11-359800 DEBUG    Read: file="html/upscalers.json" json=4 bytes=2672 time=0.000
> 08:23:11-360797 DEBUG    Read: file="extensions-builtin\sd-extension-chainner\models.json" json=24 bytes=2719 time=0.000
> 08:23:11-362792 DEBUG    chaiNNer models: path="models\chaiNNer" defined=24 discovered=0 downloaded=8
> 08:23:11-364786 DEBUG    Upscaler type=ESRGAN folder="models\ESRGAN" model="1x-ITF-SkinDiffDetail-Lite-v1"
>                          path="models\ESRGAN\1x-ITF-SkinDiffDetail-Lite-v1.pth"
> 08:23:11-365783 DEBUG    Upscaler type=ESRGAN folder="models\ESRGAN" model="4xNMKDSuperscale_4xNMKDSuperscale"
>                          path="models\ESRGAN\4xNMKDSuperscale_4xNMKDSuperscale.pth"
> 08:23:11-366781 DEBUG    Upscaler type=ESRGAN folder="models\ESRGAN" model="4x_NMKD-Siax_200k"
>                          path="models\ESRGAN\4x_NMKD-Siax_200k.pth"
> 08:23:11-369772 DEBUG    Load upscalers: total=56 downloaded=11 user=3 time=0.02 ['None', 'Lanczos', 'Nearest',
>                          'ChaiNNer', 'AuraSR', 'ESRGAN', 'LDSR', 'RealESRGAN', 'SCUNet', 'SD', 'SwinIR']
> 08:23:11-386728 DEBUG    Load styles: folder="models\styles" items=288 time=0.01
> 08:23:11-389720 DEBUG    Creating UI
> 08:23:11-390717 DEBUG    UI themes available: type=Standard themes=12
> 08:23:11-391893 INFO     UI theme: type=Standard name="black-teal"
> 08:23:11-399489 DEBUG    UI theme: css="C:\ai\automatic\javascript\black-teal.css" base="sdnext.css" user="None"
> 08:23:11-401485 DEBUG    UI initialize: txt2img
> 08:23:11-460851 DEBUG    Networks: page='model' items=71 subfolders=2 tab=txt2img folders=['models\\Stable-diffusion',
>                          'models\\Diffusers', 'models\\Reference'] list=0.04 thumb=0.00 desc=0.01 info=0.00 workers=4
>                          sort=Default
> 08:23:11-469245 DEBUG    Networks: page='lora' items=70 subfolders=0 tab=txt2img folders=['models\\Lora',
>                          'models\\LyCORIS'] list=0.04 thumb=0.01 desc=0.02 info=0.02 workers=4 sort=Default
> 08:23:11-500190 DEBUG    Networks: page='style' items=288 subfolders=1 tab=txt2img folders=['models\\styles', 'html']
>                          list=0.04 thumb=0.00 desc=0.00 info=0.00 workers=4 sort=Default
> 08:23:11-505597 DEBUG    Networks: page='embedding' items=13 subfolders=0 tab=txt2img folders=['models\\embeddings']
>                          list=0.03 thumb=0.01 desc=0.00 info=0.00 workers=4 sort=Default
> 08:23:11-507560 DEBUG    Networks: page='vae' items=0 subfolders=0 tab=txt2img folders=['models\\VAE'] list=0.00
>                          thumb=0.00 desc=0.00 info=0.00 workers=4 sort=Default
> 08:23:11-581390 DEBUG    UI initialize: img2img
> 08:23:11-822779 DEBUG    UI initialize: control models=models\control
> 08:23:12-077976 DEBUG    Read: file="ui-config.json" json=0 bytes=2 time=0.000
> 08:23:12-178414 DEBUG    UI themes available: type=Standard themes=12
> 08:23:12-704712 DEBUG    Reading failed: C:\ai\automatic\html\extensions.json [Errno 2] No such file or directory:
>                          'C:\\ai\\automatic\\html\\extensions.json'
> 08:23:12-705738 INFO     Extension list is empty: refresh required
> 08:23:13-287174 DEBUG    Extension list: processed=8 installed=8 enabled=6 disabled=2 visible=8 hidden=0
> 08:23:13-614300 DEBUG    Root paths: ['C:\\ai\\automatic']
> 08:23:13-690228 INFO     Local URL: http://127.0.0.1:7860/
> 08:23:13-691226 DEBUG    Gradio functions: registered=2361
> 08:23:13-692376 DEBUG    FastAPI middleware: ['Middleware', 'Middleware']
> 08:23:13-695397 DEBUG    Creating API
> 08:23:13-861417 INFO     [AgentScheduler] Task queue is empty
> 08:23:13-862105 INFO     [AgentScheduler] Registering APIs
> 08:23:13-980818 DEBUG    Scripts setup: ['IP Adapters:0.023', 'AnimateDiff:0.008', 'X/Y/Z Grid:0.011', 'Face:0.01']
> 08:23:13-982092 DEBUG    Model metadata: file="metadata.json" no changes
> 08:23:13-983377 DEBUG    Torch mode: deterministic=False
> 08:23:14-011305 INFO     Torch override VAE dtype: no-half set
> 08:23:14-012019 DEBUG    Desired Torch parameters: dtype=FP16 no-half=False no-half-vae=True upscast=False
> 08:23:14-013044 INFO     Setting Torch parameters: device=cuda dtype=torch.float16 vae=torch.float32 unet=torch.float16
>                          context=no_grad fp16=True bf16=None optimization=Scaled-Dot-Product
> 08:23:14-015013 DEBUG    Model requested: fn=<lambda>
> 08:23:14-016011 WARNING  Selected: model="Diffusers\Disty0/FLUX.1-dev-qint4 [82811df42b]" not found
> 08:23:14-017764 DEBUG    Script callback init time: image_browser.py:ui_tabs=0.39 system-info.py:app_started=0.06
>                          task_scheduler.py:app_started=0.13
> 08:23:14-018792 INFO     Startup time: 11.01 torch=4.11 gradio=1.27 diffusers=0.41 libraries=1.67 extensions=0.61
>                          face-restore=0.19 ui-en=0.21 ui-txt2img=0.05 ui-img2img=0.21 ui-control=0.11 ui-settings=0.23
>                          ui-extensions=1.02 ui-defaults=0.25 launch=0.13 api=0.09 app-started=0.20
> 08:23:14-020790 DEBUG    Save: file="config.json" json=35 bytes=1408 time=0.003
> 08:23:14-022783 DEBUG    Unused settings: ['cross_attention_options']

**Comment by vladmandic** [comment]
detached head can happen due to any number of reasons, but its never critical.
sdnext installer takes care of it and re-attaches it as needed. but even if it were never reattached, only impact is that such repo cannot be automatically updated.
